tensorflow/contrib/go/README.md

```diff
@@ -0,0 +1,163 @@
+# TensorFlow for Go
+
+This package provides a high-level Go API for TensorFlow, this package
+provides the necessary tools to create and manipulate Tensors, Variables,
+Constants as also to build, load and run Graphs.
+
+## API documentation
+* [Session](g3doc/session.md): Encapsulates the environment in which Operation
+  objects are executed, and Tensor objects are evaluated.
+* [Graph](g3doc/graph.md): Contains a set of Operations, which represent units
+  of computation; and Tensors, which represent the units of data that flow
+  between operations.
+* [Tensor](g3doc/tensor.md): Typed multi-dimensional array.
+
+## Practical Examples
+
+##### Python Graph generated and executed on Go
+
+This is just an example that shows how to interact with the provided API
+
+In order to generate a valid Graph you can use the next Python code
+
+```python
+import tensorflow as tf
+
+input1 = tf.placeholder(tf.int64, shape=(2, 2, 2), name='input1')
+input2 = tf.placeholder(tf.int64, shape=(2, 2, 2), name='input2')
+output = tf.add(input1, input2, name='output')
+
+with tf.Session() as sess:
+    tf.train.write_graph(sess.graph_def, '/tmp/graph/', 'test_graph.pb', as_text=True)
+```
+
+The previous code prepares two placeholders with names 'input1' and
+'input2' respectively and other tensor used as output of the addition of the
+two placeholders. At the end, it dumps the graph as text into a text file with
+path:
+'/tmp/graph/test_graph.pb'.
+
+From a Go application, you can use the next code to execute the graph:
+
+```go
+package main
+
+import (
+	"log"
+
+	"github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func main() {
+	// These are the input tensors to be used
+	inputSlice1 := [][][]int64{
+		{
+			{1, 2},
+			{3, 4},
+		}, {
+			{5, 6},
+			{7, 8},
+		},
+	}
+	inputSlice2 := [][][]int64{
+		{
+			{9, 10},
+			{11, 12},
+		}, {
+			{13, 14},
+			{15, 16},
+		},
+	}
+
+	// Create the two tensors, the data type is recognized automatically as
+	// also the tensor shape from the input slice
+	t1, err := tensorflow.NewTensor(inputSlice1)
+	if err != nil {
+		log.Fatal("Problem trying create a new tensor, Error:", err)
+	}
+
+	t2, err := tensorflow.NewTensor(inputSlice2)
+	if err != nil {
+		log.Fatal("Problem trying create a new tensor, Error:", err)
+	}
+
+	// Load the graph from the file that we generated from Python on the
+	previous step
```

Comment:
this looks like it got weirdly line-wrapped and lost its initial comment (or maybe it's just github formatting)